File: src/main/java/yte/intern/project/application/usecases/ManageEvents/Entity/Event.java
package yte.intern.project.application.usecases.ManageEvents.Entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import yte.intern.project.application.usecases.common.Entity.BaseEntity;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.Set;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@SequenceGenerator(name="idgen",sequenceName = "EVENT_SEQ")
public class Event extends BaseEntity {

    @Column( name = "NAME",unique = true)
    private String name;

    @Column( name = "MAX")
    private Long max;

    @Column(name = "CURRENT")
    private Long current;

    @Column( name = "START_DATE")
    private LocalDate startDate;

    @Column( name = "END_DATE")
    private LocalDate endDate;

    @Column( name = "START_TIME")
    private LocalTime startTime;

    @Column(name = "END_TIME")
    private LocalTime endTime;

    @Column( name = "LATITUDE")
    private Double lat;

    @Column( name = "LONGITUDE")
    private Double lng;

    @Column( name = "ASK_AGE")
    private boolean askAge;

    @Column( name = "ASK_GENDER")
    private boolean askGender;

    @OneToMany( cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "EVENT_ID")
    private Set<Question> questions;

    @OneToMany( cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "EVENT_ID")
    private Set<Participant> participants;

    @OneToMany( cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "EVENT_ID")
    private Set<SurveyEventQuestion> surveyQuestions;

}

File: FlutterMobileExtension/lib/models/Event.dart
import 'package:flutter/material.dart';
import 'package:tubitak_intern_mobile_project/models/Question.dart';
import 'package:tubitak_intern_mobile_project/models/SurveyQuestion.dart';

class Event {
  final String name;
  final int max;
  final int current;
  final String startDate;
  final String startTime;
  final String endDate;
  final String endTime;
  final bool askAge;
  final bool askGender;
  List<Question> questions;
  List<SurveyQuestion> surveyQuestions;

  Event(
      {this.name,
      this.max,
      this.current,
      this.startDate,
      this.startTime,
      this.endDate,
      this.endTime,
      this.questions,
      this.askAge,
      this.askGender,
      this.surveyQuestions});

  factory Event.fromJson(Map<String, dynamic> json) {
    var questionObjsJson = json['questions'] as List;
    List<Question> questionObjs =
        questionObjsJson.map((tagJson) => Question.fromJson(tagJson)).toList();

    var surveyQuestionObjsJson = json['surveyQuestions'] as List;
    List<SurveyQuestion> surveyQuestionObjs = surveyQuestionObjsJson
        .map((tagJson) => SurveyQuestion.fromJson(tagJson))
        .toList();

    return Event(
        name: json['name'],
        max: json['max'],
        current: json['current'],
        startDate: json['startDate'],
        startTime: json['startTime'],
        endDate: json['endDate'],
        endTime: json['endTime'],
        askAge: json['askAge'],
        askGender: json['askGender'],
        questions: questionObjs,
        surveyQuestions: surveyQuestionObjs);
  }
}
